src/core/dfn-abbr.js

```diff
@@ -0,0 +1,48 @@
+// @ts-check
+// Module core/dfn
+// - Finds all <dfn> elements with data-abbr attribute and parses them and registers definitions.
+// - Also populates abbrMap used by core/inlines for such definitions.
+
+import { getAbbreviationFromText } from "./utils";
+import { registerDefinition } from "./dfn-map";
+
+export const name = "core/dfn-abbr";
+export const abbrMap = new Map();
+
+/**
+ * Parses Elements of form <dfn data-abbr="PAN">Permanent Account Number</dfn>
+ * to <dfn data-abbr="PAN">Permanent Account Number (PAN)</dfn>
+ * @param {HTMLElement} dfn Element to be parsed.
+ */
+function parseAbbreviatedDefinition(dfn) {
+  // checks if text content is already in the form Permanent Account Number (PAN)
+  const matched = /\((.*?)\)/.exec(dfn.textContent);
+  if (!matched || !(matched.pop() === dfn.dataset.abbr))
+    dfn.textContent = dfn.textContent.concat(` (${dfn.dataset.abbr})`);
+}
+
+export function run() {
+  document.querySelectorAll("dfn").forEach(dfn => {
```

Comment:
If I use ```document.querySelectorAll("dfn[data-abbr]")```, I run into jsDoc issues. Apparently unlike other ```querySelectorAll``` functions,  ```document.querySelectorAll("tag[attr]")``` returns an Element, and not an HTMLElement, which does not have several properties used (like dataset)

Reply:
@marcoscaceres is it fine if we leave it this way?

Reply:
document.querySelectorAll returns a NodeList. What you are describing doesn't sound correct (or is a bug in TS... which would also be odd, given that document.querySelectorAll is used all the time, even in our code base)

Reply:
cc @saschanaz - might know what's going on from a typescript perspective

Reply:
You'll want to do this way:

```js
/** @type {NodeListOf<HTMLElement>} */
const dfns = document.querySelectorAll("dfn[data-abbr]");
dfns.forEach(dfn => { ... });
```

TS knows `dfn` is an `HTMLElement` but starts to be confused when there is any more selector thing.

Reply:
Interesting. Breaking it up that way works. Thanks :smile: